DependencyInjectionContainer/DIContainerBuilder.cs

```diff
@@ -0,0 +1,124 @@
+﻿namespace DependencyInjectionContainer;
+using Attributes;
+using Enums;
+using Exceptions;
+using System.Reflection;
+
+public sealed class DIContainerBuilder
+{
+    private List<Service> services = new List<Service>();
+    private DIContainer? parentContainer;
+    private bool isBuild;
+
+    public DIContainerBuilder() { }
+    internal DIContainerBuilder(DIContainer parent) => parentContainer = parent;
+
+    public void Register<TImplementationInterface, TImplementation>
+        (ServiceLifetime lifetime, Func<DIContainer, TImplementation> implementationFactory = null) where TImplementation : TImplementationInterface
```

Comment:
why you allow parameter to be null?

![image](https://user-images.githubusercontent.com/8506504/216564044-6d3a021c-35ec-4096-9ab7-28828b74c277.png)